src/pandablocks_ioc/_pvi.py

```diff
@@ -278,6 +295,10 @@ def add_pvi_info(record_name: EpicsName, group: PviGroup, component: Component):
     def add_general_device_refs_to_groups(device: Device):
         for group in device.children:
             if group.name in Pvi._general_device_refs:
+                if not isinstance(group, Group):
+                    raise RuntimeError(
+                        "Tried to add a .... to a " f"widget group {group} ... "
+                    )
                 group.children.append(Pvi._general_device_refs[group.name])
```

Comment:
I think the last assignment here will be seen as type safe because we'll be allowed to use a list for `Sequence[ComponentUnion]`, if not put it back to an append and add `# type: ignore`.

```suggestion
        for device_child in device.children:
            if device_child.name in Pvi._general_device_refs:
                if not isinstance(device_child, Group):
                    raise RuntimeError(
                        "Tried to add a .... to a " f"widget group {device_child} ... "
                    )
                device_child.children = (
                    list(device_child.children)
                    + [Pvi._general_device_refs[device_child.name]]
                )
```